docs/path-resolution.rst

```diff
@@ -0,0 +1,470 @@
+    :ref:`base path <base-path>` and :ref:`import remapping <import-remapping>`.
+
+.. index:: ! base path, --base-path
+.. _base-path:
+
+Base Path
+=========
+
+Base path specifies the directory that the Host Filesystem Loader will load files from.
+It is simply prepended to a source unit name before the filesystem lookup is performed.
+
+By default base path is empty, which leaves the source unit name unchanged.
+When the source unit name is a relative path, this results in the file being looked up in the
+directory the compiler has been invoked from.
+It is also the only value that results in absolute paths in source unit names being actually
+interpreted as absolute paths on disk.
+
+If the base path itself is relative, it is also interpreted as relative to the current working
+directory of the compiler.
+
+.. index:: ! remapping; import, ! import; remapping, ! remapping; context, ! remapping; prefix, ! remapping; target
+.. _import-remapping:
+
+Import Remapping
+================
+
+Import remapping allows you to redirect imports to a different location in the virtual filesystem.
+The mechanism works by changing the translation between import paths and source unit names.
+For example you can set up a remapping so that any import from the virtual directory
+``github.com/ethereum/dapp-bin/library/`` would be seen as an import from ``dapp-bin/library/`` instead.
+
+You can limit the scope of a remapping by specifying a *context*.
+This allows creating remappings that apply only to imports located in a specific library or a specific file.
+Without a context a remapping is applied to every matching import in all the files in the virtual
+filesystem.
+
+Import remappings have the form of ``context:prefix=target``:
+
+- ``context`` must match the beginning of the source unit name of the file containing the import.
+- ``prefix`` must match the beginning of the source unit name resulting from the import.
+- ``target`` is the value the prefix is replaced with.
+
+For example, if you clone https://github.com/ethereum/dapp-bin/ locally to ``/project/dapp-bin``
+and run the compiler with:
+
+.. code-block:: bash
+
+    solc github.com/ethereum/dapp-bin/=dapp-bin/ --base-path /project source.sol
+
+you can use the following in your source file:
+
+.. code-block:: solidity
+
+    import "github.com/ethereum/dapp-bin/library/math.sol"; // source unit name: dapp-bin/library/math.sol
+
+The compiler will look for the file in the VFS under ``dapp-bin/library/math.sol``.
+If the file is not available there, the source unit name will be passed to the Host Filesystem
+Loader, which will then look in ``/project/dapp-bin/library/iterable_mapping.sol``.
+
+.. warning::
+
+    Information about remappings is stored in contract metadata.
+    Since the binary produced by the compiler has a hash of the metadata embedded in it, any
+    modification to the remappings will result in a slightly different bytecode.
```

Comment:
```suggestion
    modification to the remappings will result in different bytecode.
```

Reply:
Well, it's only the tiny bit containing the metadata hash that is different :) I did not want to give an impression that it changes the behavior of the contract.